### PR TITLE
Various codex improvements and bugfixes.

### PR DESCRIPTION
**Pull request by MistakeNot4892**
## Description of changes
- The codex now uses span classes for formatting.
- Codex pages in a category now show their categories in a footer.
- Codex internals are a bit nicer to override/invoke (body, header, footer)
- Various issues with codex name sanitization and lookup resolved.
- Unit test added to validate codex links.
- Fixed some redundant 'EMP Pulse' usage.
- Nutriment now populates the codex again.

## Why and what will this PR improve
Codex work better.

## Authorship
Myself.

## Changelog
Nothing player-facing.

### CODE REVIEW

File: code/_helpers/text.dm [hunk 1]
```diff
@@ -319,6 +319,25 @@
 /proc/trim(text)
 	return trim_left(trim_right(text))
 
+// Remove starting and ending <br> and variants
+/proc/trim_trailing_linebreaks(var/text)
```
Comment (afterthought2) [inline code/_helpers/text.dm]: I feel running a regex would be a more sane implementation.

File: code/_helpers/text.dm [hunk 2]
```diff
@@ -463,7 +482,7 @@
  * Strip out the special beyond characters for \proper and \improper
  * from text that will be sent to the browser.
  */
-#define strip_improper(input_text) replacetext(replacetext(input_text, "\proper", ""), "\improper", "")
+#define strip_improper(input_text) replacetext(replacetext(replacetext(input_text, "\proper", ""), "\improper", ""), "ÿ", "")
```
Comment (afterthought2) [inline code/_helpers/text.dm]: Unclear that this will work, as per discord.

File: code/modules/codex/entries/_codex_entry.dm [hunk 3]
```diff
@@ -33,48 +34,69 @@
...
-	dat += "<hr><h2>[name]</h2>"
-	return jointext(dat, null)
 
-/datum/codex_entry/proc/get_text(var/mob/presenting_to)
-	var/list/dat = list(get_header(presenting_to))
+/datum/codex_entry/proc/get_codex_header(var/mob/presenting_to)
+	. = list()
+	if(presenting_to)
+		var/datum/codex_entry/linked_entry = SScodex.get_entry_by_string("nexus")
+		. += "<a href='?src=\ref[SScodex];show_examined_info=\ref[linked_entry];show_to=\ref[presenting_to]'>Home</a>"
```
Comment (afterthought2) [inline code/modules/codex/entries/_codex_entry.dm:66]: Unless you are actually doing something nontrivial with `show_to`, I think it's better practice to infer it from `usr` in `Topic`.

Comment (MistakeNot4892) [reply code/modules/codex/entries/_codex_entry.dm:66]: I would prefer not to infer from `usr`. This is accessed from a few places, including via a client verb which would potentially confuse things.

File: code/_helpers/text.dm (hunk 2 above)
Comment (MistakeNot4892) [reply code/_helpers/text.dm]: Reverted this, will revisit it in the codex dump PR.

File: code/_helpers/text.dm (hunk 1 above)
Comment (MistakeNot4892) [reply code/_helpers/text.dm]: Regex added.

**Review by afterthought2 — DISMISSED** [review]

**Review by afterthought2 — APPROVED** [review]

**Pull request merged**